src/ostree/ot-builtin-commit.c

```diff
@@ -631,12 +631,12 @@ ostree_builtin_commit (int argc, char **argv, GCancellable *cancellable, GError
                   if (!ostree_repo_import_archive_to_mtree (repo, &opts, archive, mtree,
                                                             modifier, cancellable, error))
                     goto out;
-                }
 #else
               g_set_error (error, G_IO_ERROR, G_IO_ERROR_NOT_SUPPORTED,
                            "This version of ostree is not compiled with libarchive support");
               return FALSE;
 #endif
+                }
```

Comment:
Note that this was fixed in https://github.com/ostreedev/ostree/pull/1181, so you should be able to just rebase and drop that one out.